content/blog/2022/10/envvar-policy.md

```diff
@@ -0,0 +1,146 @@
+---
+title: Enforcing compliance in your environment variables
+authors:
+- José Guilherme Vanz
+date: 2022-11-01
+---
+
+We're glad to present to Kubewarden users the new [environment-variable-policy](https://github.com/kubewarden/environment-variable-policy). It's able to inspect containers, init containers, and ephemeral containers, and restricts their usage of environment variables by checking their names and values defined under the containers' [`env[*]`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#envvar-v1-core).
+
+As always, the policy can be found in [Github](https://github.com/kubewarden/environment-variable-policy) and all the  artifacts, including the BOM files, are signed with Sigstore.
+
+## What is so useful about the new environment-variable policy?
+
+This policy complements the recently released  [env-variable-secrets-scanner](https://www.kubewarden.io/blog/2022/10/env-var-secrets/) policy. Both policies focus on validating the environment values provided to Kubernetes Pod objects. In this new policy, users can validate which variables name and values their resources should have. For that, the policy gives four rule operators to control the environment variable defined in the containers. They are:
+
+### anyIn
+
+`anyIn` works as a deny list of your usage of volumes. It checks if any of the `environmentVariables` are in the Pod/Workload resource:
+```yaml
+settings:
+  rules:
+    - reject: anyIn
+      environmentVariables:
+        - name: "envvar1"
+        - name: "envvar2"
+```
+
+### anyNotIn
+
+`anyNotIn` works as an allow list. Checks if any of the `environmentVariables` are not in the Pod/Workload resource:
+
+```yaml
+settings:
+  rules:
+    - reject: anyNotIn
+      environmentVariables:
+        - name: "envvar2"
+          value: "envvar2_value"
+        - name: "envvar3"
+```
+
+### allAreUsed
+
+`allAreUsed` the container cannot use all listed volumes at once. Checks if all of the `environmentVariables` are in the Pod/Workload resource:
+
+```yaml
+settings:
+  rules:
+    - reject: allAreUsed
+      environmentVariables:
+        - name: "envvar2"
+          value: ""
+```
+
+### notAllAreUsed
+
+`notAllAreUsed` the container can use all listed volumes at once, but only one of them. Checks if all of the `environmentVariables` are not in the Pod/Workload resource
+
+```yaml
+settings:
+  rules:
+    - reject: notAllAreUsed
+      environmentVariables:
+        - name: "envvar3"
+          value: "envvar3_value"
+        - name: "envvar4"
+          value: "envvar4_value"
+```
+
+
+The user is allowed to define multiple rules in the policy settings. The resource must be approved by all of them in order to be accepted into the cluster.
+
+## An example
```

Comment:
Can we move that example right into the `allAreUsed` section and add similar ones for each rule?

I understand this would grow the article a lot, maybe we can - for the yet to be written examples - use something like that:

> For example, given the following configuration:
> 
> ```yaml
> settings:
>   rules:
>     - reject: anyIn
>       environmentVariables:
>         - name: "foo"
>           value: "bar"
>         - name: "foo2"
>           value: "bar2"
> ```
> 
> The following Pod would be rejected:
> 
> ```yaml
> apiVersion: v1
> kind: Pod
> metadata:
>   name: nginx
>   spec:
>     containers:
>     - name: nginx
>       image: nginx:1.14.2
>       ports:
>       - containerPort: 80
>       env:
>         - name: "foo"
>           value: "bar"
> ```

Some notes about that:

  * Save space by not showing `kubectl` invocations
  * Save space by showing only the settings of the policy, not the whole `ClusterAdmissionPolicy` definition
  * Save space by showing just the `Pod` definition, which is shorter than `Deployment` or other things

It's however nice to show that the policy can work also on `Deployment` resources, hence do not overwrite the current example.